trieste/acquisition/function/function.py

```diff
@@ -223,12 +232,132 @@ def __call__(self, x: TensorType) -> TensorType:
         return (self._eta - mean) * normal.cdf(self._eta) + variance * normal.prob(self._eta)
 
 
+class log_expected_improvement(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, eta: TensorType):
+        r"""
+        Return the Log Expected Improvement (LogEI) acquisition function for single-objective global
+        optimization. Improvement is with respect to the current "best" observation ``eta``, where
+        an improvement moves towards the objective function's minimum and the expectation is
+        calculated with respect to the ``model`` posterior. LogEI mitigates the issue of
+        Expected Improvement (EI) that it is numerically zero in many regions.
+        Although LogEI has (approximately) same optima as EI, it is much easier to optimze,
+        and empirically shows significantly improved optimization performance.
+        See :cite: `ament2023unexpected` for details.
+
+        :param model: The model of the objective function.
+        :param eta: The "best" observation.
+        :return: The expected improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
```

Comment:
is check_shapes actually raising this at runtime?
I vaguely recall that it doesn't do that...

Reply:
No, it raises `StageError`, so I removed it in https://github.com/secondmind-labs/trieste/pull/904/commits/0cec4201066bbdf89577739a0723d9c21816304b and do shape check by `tf.debugging.assert_shapes`